internal/tools/cypher/get_schema_handler.go

```diff
@@ -54,12 +58,164 @@ func handleGetSchema(ctx context.Context, deps *tools.ToolDependencies) (*mcp.Ca
 		slog.Warn("schema is empty, no data in the database")
 		return mcp.NewToolResultText("The get-schema tool executed successfully; however, since the Neo4j instance contains no data, no schema information was returned."), nil
 	}
-	// Convert records to JSON using the existing utility function
-	response, err := deps.DBService.Neo4jRecordsToJSON(records)
-
+	structuredOutput, err := processCypherSchema(records)
+	if err != nil {
+		slog.Error("failed to process get-schema Cypher Query", "error", err)
+		return mcp.NewToolResultError(err.Error()), nil
+	}
+	jsonData, err := json.Marshal(structuredOutput)
 	if err != nil {
-		slog.Error("failed to format schema results to JSON", "error", err)
+		slog.Error("failed to serialize structured schema", "error", err)
 		return mcp.NewToolResultError(err.Error()), nil
+
+	}
+	return mcp.NewToolResultText(string(jsonData)), nil
+}
+
+type SchemaItem struct {
+	Key   string       `json:"key"`
+	Value SchemaDetail `json:"value"`
+}
+
+type SchemaDetail struct {
+	Type          string                  `json:"type"`
+	Properties    map[string]string       `json:"properties,omitempty"`
+	Relationships map[string]Relationship `json:"relationships,omitempty"`
+}
+
+type Relationship struct {
+	Direction  string            `json:"direction"`
+	Labels     []string          `json:"labels"` // List of target node labels
+	Properties map[string]string `json:"properties,omitempty"`
+}
+
+// processCypherSchema is a func that transform a list of Neo4j.Record in a JSON tagged struct,
+// this allow to maintain the same APOC query supported by multiple Neo4j versions while returning a tokens aware version of it.
```

Comment:
```suggestion
// this allows us to maintain the same APOC query supported by multiple Neo4j versions while returning a tokens aware version of it.
```

Unsure of this correction tbh.